--- "a/src/beman/exemplar/CMakeLists.txt"	2025-02-20 15:57:28.000000000 -0500
+++ "b/src/beman/exemplar/CMakeLists.txt"	2025-05-09 18:59:51.279809525 -0400
@@ -1,25 +1,42 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception

-add_library(beman.exemplar)
+add_library(beman.exemplar INTERFACE)
 add_library(beman::exemplar ALIAS beman.exemplar)

-target_sources(beman.exemplar PRIVATE identity.cpp)
-
 target_sources(
     beman.exemplar
-    PUBLIC
+    INTERFACE
         FILE_SET HEADERS
         BASE_DIRS ${PROJECT_SOURCE_DIR}/include
         FILES ${PROJECT_SOURCE_DIR}/include/beman/exemplar/identity.hpp
 )

-set_target_properties(beman.exemplar PROPERTIES VERIFY_INTERFACE_HEADER_SETS ON)
+set_target_properties(beman.exemplar PROPERTIES VERIFY_INTERFACE_HEADER_SETS ON
+    EXPORT_NAME exemplar
+)

 install(
     TARGETS beman.exemplar
     EXPORT beman.exemplar
-    DESTINATION
-    $<$<CONFIG:Debug>:debug/>${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION $<$<CONFIG:Debug>:debug/>${CMAKE_INSTALL_BINDIR}
-    FILE_SET HEADERS DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+    FILE_SET HEADERS
+)
+
+include(CMakePackageConfigHelpers)
+
+write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/beman.exemplar-version.cmake"
+    VERSION 1.0.0
+    COMPATIBILITY ExactVersion
+)
+
+install(
+    FILES "${CMAKE_CURRENT_BINARY_DIR}/beman.exemplar-version.cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/beman.exemplar"
+)
+
+install(
+    EXPORT beman.exemplar
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/beman.exemplar"
+    NAMESPACE beman::
+    FILE beman.exemplar-config.cmake
 )
