--- a/src/beman/exemplar/CMakeLists.txt	2025-03-05 06:01:47.000000000 -0500
+++ b/src/beman/exemplar/CMakeLists.txt	2025-05-09 19:34:26.861986882 -0400
@@ -2,14 +2,12 @@

 include(GNUInstallDirs)

-add_library(beman.exemplar)
+add_library(beman.exemplar INTERFACE)
 add_library(beman::exemplar ALIAS beman.exemplar)

-target_sources(beman.exemplar PRIVATE identity.cpp)
-
 target_sources(
     beman.exemplar
-    PUBLIC
+    INTERFACE
         FILE_SET HEADERS
         BASE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../../../include
         FILES
